specifyweb/frontend/js_src/lib/components/Statistics/StatsSpec.tsx

```diff
@@ -0,0 +1,383 @@
+import { getDateInputValue } from '../../utils/dayJs';
+import type { IR } from '../../utils/types';
+import { formatNumber } from '../Atoms/Internationalization';
+import { queryFieldFilters } from '../QueryBuilder/FieldFilter';
+import type { BackendStatsResult, BackendStat, StatItemSpec } from './utils';
+
+function modifyBackendResult<CATEGORY_NAME extends keyof BackendStatsResult>(
+  backendobject: BackendStatsResult[CATEGORY_NAME] | undefined,
+  modifyfunction: (
+    rawValue: BackendStatsResult[CATEGORY_NAME][keyof BackendStatsResult[CATEGORY_NAME]]
+  ) => BackendStat['value']
+): StatCategoryReturn {
+  if (backendobject === undefined) {
+    return undefined;
+  }
+  return Object.fromEntries(
+    Object.entries(backendobject).map(([key, value]) => [
+      key,
+      {
+        label: key,
+        spec: {
+          type: 'Backendstat',
+          value: modifyfunction(value),
+        },
+      },
+    ])
+  );
+}
+
+export type StatCategoryReturn =
+  | IR<{ readonly label: string; readonly spec: StatItemSpec }>
+  | undefined;
+
+type StatsSpec =
+  | {
+      readonly [CATEGORY_NAME in keyof BackendStatsResult]: {
+        readonly label: string;
+        readonly categories: (
+          backendStatsResult: BackendStatsResult[CATEGORY_NAME] | undefined
+        ) => StatCategoryReturn;
+      };
+    }
+  | {
+      readonly [CATEGORY_NAME in string]: {
+        readonly label: string;
+        readonly categories: () => StatCategoryReturn;
+      };
+    };
+
+export const statsSpec: StatsSpec = {
+  holdings: {
+    label: 'Holdings',
+    categories(backendStatsResult) {
+      return {
+        specimen: {
+          label: 'Specimens',
```

Comment:
Need to extract the labels into a localization file.

I think it is worth creating a new file in the `./localization` directory that would contain all localization strings specific to the Stats page (because there would be a lot of them).

To do that, just copy some existing file in the `./localization` directory, call it `stats.tsx`, and rename the `export const someText` to `export const statsText` (near the top of the file).

Then remove everything after `createDictionary({` (delete previous localization strings from that file) and put the new localization strings in their place